scipy/signal/_spectral_py.py

```diff
@@ -243,11 +243,11 @@ def lombscargle(
         )
 
     # weight vector must sum to 1
-    weights *= 1.0 / weights.sum()
+    weights = weights * (1.0 / weights.sum())
```

Comment:
It may also be sensible to add a regression test for this mutation fix as well--the one I just suggested doesn't fail if I revert this part of the patch of course.

We can handle that if you don't want to, should be straightforward/similar to the other one but for `weights`.